### PR TITLE
Fix loading of ONNX models with Resize operation with Opset 11 for newer versions of Pytorch

### PR DESCRIPTION
**Pull request by sl-sergei**
With Pytorch ver. 1.6 (or newer) onnx importer with Opset 11 stores **scales** parameter as float32.

relates: https://github.com/opencv/opencv/issues/18253 and https://github.com/opencv/opencv/issues/18300

merge with extra: opencv/opencv_extra#806

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

<cut/>

```
opencv_extra=testdata_for_resize_opset11
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.4.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Review by dkurt — APPROVED** [review]
:+1: Thank you!

**Comment by asmorkalov** [comment]
@mshabunin could you merge the patch?

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -1557,7 +1557,9 @@ void ONNXImporter::populateNet(Net dstNet)
             Mat shapes = getBlob(node_proto, constBlobs, node_proto.input_size() - 1);
             CV_CheckEQ(shapes.size[0], 4, "");
             CV_CheckEQ(shapes.size[1], 1, "");
-            CV_CheckTypeEQ(shapes.depth(), CV_32S, "");
+            CV_Check(shapes.depth(), shapes.depth() == CV_32S || shapes.depth() == CV_32F, "");
```
Comment (alalek) [inline modules/dnn/src/onnx/onnx_importer.cpp:1560]:
```.diff
-CV_Check
+CV_CheckDepth
```

for better diagnostic messages.

**Comment by berak** [comment]
thanks ! using your fix,  https://github.com/princeton-vl/YouTube3D works now with opencv ;)

**Comment by dkurt** [comment]
> relates: #18253 and #18300

please close if the issues have been resolved

**Pull request merged**